### PR TITLE
Add entity and throttle utilization stats relative to configurable limits

### LINKED ISSUE #3583: Add several `*PercentUsed` gauge stats
### Problem

As a node operator, I would like more visibility into the status of my node's throttles and overall trends in network utilization.

### Solution

Add,

1️⃣ "Frontend" gauges that measure HAPI throttles:
  - `throughputLimitsPercentUsed`
  - `priorityReservationsPercentUsed`
  - `creationLimitsPercentUsed`
  - `freeQueryLimitsPercentUsed`
  - `offHeapQueryLimitsPercentUsed`

2️⃣ "Backend" gauges that measure network utilization:
  - `accountsPercentCreated`
  - `consensusGasPercentUsed`
  - `contractsPercentCreated`
  - `filesPercentCreated`
  - `nftsPercentMinted`
  - `topicsPercentCreated`
  - `tokensPercentCreated`
  - `scheduledTransactionsPercentCreated`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issues**:
 - Closes #3583 

**Description**:
Adds `DoubleGauge` statistics to track the percent usage of specified throttles, both HAPI and consensus; and adds gauges to track the percent usage of system entity limits.

This PR adds and enforces these limits for accounts, contracts, files, scheduled transactions, tokens, token associations, and topics; standardizing with the existing NFT and contract storage limits in a new `UsageLimits` singleton.

1️⃣ _Throttle usage statistics_ 
New `DoubleGauge` stats, updated every `stats.throttleUtils.gaugeUpdateIntervalMs=1000` milliseconds [here](https://github.com/hashgraph/hedera-services/blob/03583-add-utilization-stats/hedera-node/src/main/java/com/hedera/services/stats/ServicesStatsManager.java#L112):
 - `consGasPercentUsed`: instantaneous % used in cons Gas throttle bucket
 - `consThroughputLimitsPercentUsed`: instantaneous % used in cons ThroughputLimits throttle bucket
 - `hapiFreeQueryLimitsPercentUsed`: instantaneous % used in HAPI FreeQueryLimits throttle bucket
 - `hapiGasPercentUsed`: instantaneous % used in HAPI Gas throttle bucket
 - `hapiThroughputLimitsPercentUsed`: instantaneous % used in HAPI ThroughputLimits throttle bucket

2️⃣  _System entity limits_
All global/dynamic properties,
```
accounts.maxNumber=5_000_000
contracts.maxKvPairs.aggregate=500_000_000
contracts.maxNumber=5_000_000
files.maxNumber=1_000_000
scheduling.maxNumber=10_000_000
tokens.maxAggregateRels=10_000_000
tokens.maxNumber=1_000_000
tokens.nfts.maxAllowedMints=5000000
topics.maxNumber=1_000_000
```

3️⃣ _Entity usage statistics_ 
New `DoubleGauge` stats, updated every `stats.entityUtils.gaugeUpdateIntervalMs=3000` milliseconds:
 - `accountsPercentUsed`: instantaneous % used of accounts system limit
 - `contractsPercentUsed`: instantaneous % used of contracts system limit
 - `filesPercentUsed`: instantaneous % used of files system limit
 - `nftsPercentUsed`: instantaneous % used of NFTs system limit
 - `schedulesPercentUsed`: instantaneous % used of schedules system limit
 - `storageSlotsPercentUsed`: instantaneous % used of storage slots system limit
 - `tokensPercentUsed`: instantaneous % used of tokens system limit
 - `tokenRelsPercentUsed`: instantaneous % used of token associations system limit
 - `topicsPercentUsed`: instantaneous % used of topics system limit

### CODE REVIEW

File: hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java [hunk 1]
```diff
@@ -85,4 +91,15 @@ void reclaimLastAllowedUse() {
     public DiscreteLeakyBucket bucket() {
         return bucket;
     }
+
+	private long effectiveLeak(final long elapsedNanos) {
+		if (elapsedNanos >= TIME_TO_EMPTY) {
+			return bucket.totalCapacity();
+		} else {
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java:105]: I'm not sure why that check was there 🤔; it's safe to leak 0 nanos worth of capacity.

File: hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java [hunk 2]
```diff
@@ -180,18 +185,21 @@ public TransactionalLedger<Pair<AccountID, TokenID>, TokenRelProperty, MerkleTok
 	public void begin() {
 		autoCreationLogic.reset();
 		accountsLedger.begin();
-		mutableEntityAccess.begin();
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java:183]: `MutableEntityAccess` had a counter-intuitive responsibility for managing the `tokensLedger` lifecycle; moved that responsibility into `HederaLedger`.

**Review by statop — APPROVED** [review]
Assuming the comment about negative values is resolved, lgtm :-)

File: hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java:105]: The `DiscreteLeakyBucket` rejects negative values [here](https://github.com/hashgraph/hedera-services/blob/master/hapi-utils/src/main/java/com/hedera/services/throttles/DiscreteLeakyBucket.java#L61).

**Pull request merged**